src/realm/sync/config.hpp

```diff
@@ -224,6 +224,9 @@ struct SyncConfig {
 
     bool simulate_integration_error = false;
 
+    std::function<void(std::shared_ptr<Realm>)> subscription_initializer{nullptr};
+    bool always_run = false;
```

Comment:
Yep. I am not sure if I need this flag, because detecting whether is the first time that the file is open or not can be tricky. I don't think that checking if the file is present at that specific path is enough... 
This is the original request:

> This callback have to be called only if the realm file is created for the first time.
> There should be also a configuration option rerunOnOpen. If it is set to true the InitialSubscriptionsCallback should be called each time an application that is using realm SDK is launched.
> Consider running two applications using the same Realm file and running one application twice.
>

Reply:
I am removing this boolean for now. Since there is no clear way of detecting if realm is opened for the first time. Once we have a mechanism for it, adding the logic for it, won't be that difficult. 
Ref: https://github.com/realm/realm-core/issues/6734